### PR TITLE
Optimize getEntryLogMetadata

### PR DESCRIPTION
**Pull request by gaozhangmin**
In the `getEntryLogMetadata` method, the `FileNotFoundException` exception is caught before the more general Exception. This modification ensures that if the log file is not found, there is no need to proceed with the `extractEntryLogMetadataByScanning(entryLogId, throttler)` method call.

This enhancement improves the code's error handling by distinguishing between different exceptions that may occur during the retrieval of entry log metadata. It ensures that unnecessary operations, scanning the log, are skipped when the log file is not found.

### CODE REVIEW

**Comment by gaozhangmin** [comment]
rerun failure checks

**Comment by gaozhangmin** [comment]
rerun failure checks

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/DefaultEntryLogger.java [hunk 1]
```diff
@@ -1036,6 +1036,9 @@ public EntryLogMetadata getEntryLogMetadata(long entryLogId, AbstractLogCompacto
         // entry log
         try {
             return extractEntryLogMetadataFromIndex(entryLogId);
+        } catch (FileNotFoundException ex) {
```
Comment (hangc0276) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/DefaultEntryLogger.java:1039]: This exception will be thrown by `extractEntryLogMetadataByScanning`, and I don't think it is necessary to catch the exception here.

Comment (gaozhangmin) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/DefaultEntryLogger.java:1039]: If  `FileNotFoundException` is caught,  no need to proceed with the `extractEntryLogMetadataByScanning(entryLogId, throttler)` method call.  save unnecessary call.

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/DefaultEntryLogger.java [hunk 2]
```diff
@@ -1036,6 +1036,9 @@ public EntryLogMetadata getEntryLogMetadata(long entryLogId, AbstractLogCompacto
...
+            LOG.warn("Cannot find entry log file {}.log", entryLogId);
+            throw new IOException(ex);
```
Comment (horizonzy) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/DefaultEntryLogger.java:1041]: Why wrap the FileNotFoundExpcetion.

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/DefaultEntryLogger.java [hunk 3]
```diff
@@ -1036,6 +1036,9 @@ public EntryLogMetadata getEntryLogMetadata(long entryLogId, AbstractLogCompacto
         // entry log
         try {
             return extractEntryLogMetadataFromIndex(entryLogId);
+        } catch (FileNotFoundException ex) {
+            LOG.warn("Cannot find entry log file {}.log", entryLogId);
```
Comment (horizonzy) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/DefaultEntryLogger.java:1040]: The file entryLogId should be hex, we can get the log info by ex.getMessage()

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/DefaultEntryLogger.java (hunk 2 above)
Comment (gaozhangmin) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/DefaultEntryLogger.java:1041]: Please read the context above. @horizonzy

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/DefaultEntryLogger.java [hunk 4]
```diff
@@ -1036,6 +1036,9 @@ public EntryLogMetadata getEntryLogMetadata(long entryLogId, AbstractLogCompacto
         // entry log
         try {
             return extractEntryLogMetadataFromIndex(entryLogId);
+        } catch (FileNotFoundException fne) {
+            LOG.warn("Cannot find entry log file {}.log : {}", entryLogId, fne.getMessage());
```
Comment (horizonzy) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/DefaultEntryLogger.java:1040]: `{}.log` format should be hex. EntryLogId is decimal.

**Comment by gaozhangmin** [comment]
rerun failure checks

**Review by wenbingshen — APPROVED** [review]

**Review by hangc0276 — APPROVED** [review]

**Review by horizonzy — APPROVED** [review]
LGTM.

**Review by zymap — APPROVED** [review]

**Comment by gaozhangmin** [comment]
rerun failure checks

**Comment by gaozhangmin** [comment]
rerun failure checks

**Review by StevenLuMT — APPROVED** [review]

**Pull request merged**